actix-web-codegen/src/route.rs

```diff
@@ -96,10 +102,50 @@ impl Args {
                                 "Attribute wrap expects type",
                             ));
                         }
+                    } else if nv.path.is_ident("method") {
+                        if let syn::Lit::Str(ref lit) = nv.lit {
```

Comment:
Would it make sense to provide an implementation of `TryFrom<&syn::Lit> for GuardType`?

With that, the code here will be simplified to just:

```rust
let guard = GuardType::try_from(&nv.lit)?;
if !methods.insert(guard) { ... };
```

Reply:
I managed this:
```rust
                    } else if nv.path.is_ident("method") {
                        if let syn::Lit::Str(ref lit) = nv.lit {
                            let guard = GuardType::try_from(lit)?;
                            if !methods.insert(guard) {
                                return Err(syn::Error::new_spanned(
                                    &nv.lit,
                                    &format!(
                                        "HTTP Method defined more than once: `{}`",
                                        lit.value()
                                    ),
                                ));
                            }
                        } else {
                            return Err(syn::Error::new_spanned(
                                nv.lit,
                                "Attribute method expects literal string!",
                            ));
                        }
                    } else {
``` 
So doing the conversation from `&syn::LitStr` otherwise using `&syn::Lit` would mean having the `"Attribute method expects literal string!"` error inside the `GuardType` conversion which seems wrong.

Reply:
Makes sense.